src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -257,10 +257,10 @@ private static int GetSegmentCountIfCombined(ArrayBuilder<SourceText> segments,
                     int count = 1;
                     for (int j = i + 1; j < segments.Count; j++)
                     {
-                        if (segments[j].Length <= segmentSize)
-                        {
-                            count++;
-                        }
+                        if (segments[j].Length > segmentSize)
```

Comment:
Are these now just stylistic changes, rather than perf improvements? If so, consider reverting.

Reply:
do you mean the braces? If so, I can add those back. As for inverting the condition and breaking, it was intentional to change this code to stop matching once it was out of a contiguous matching sequence.